test/deriving/test.ml

```diff
@@ -40,3 +54,29 @@ type nonrec int = int [@@deriving foo, bar]
 type nonrec int = int
 val x : int = 42
 |}]
+
+module Foo_sig : sig
+  type t [@@deriving foo]
+end = struct
+  type t
+end
+[%%expect{|
+Line _, characters 25-25:
+Error: ppxlib: [@@@deriving.end] attribute missing
```

Comment:
Isn't it a bit weird that this error persists? I'd expect an error saying the module doesn't provide the `val y : int` required by the signature.

We should probably have a look at that separately